lib/rubocop/cop/rspec/factory_bot/dynamic_attribute_defined_statically.rb

```diff
@@ -41,15 +41,20 @@ class DynamicAttributeDefinedStatically < Cop
           def on_block(node)
             return if node.method_name == :trait
             factory_attributes(node).to_a.flatten.each do |attribute|
-              if dynamic_defined_statically?(attribute)
-                add_offense(attribute, location: :expression)
-              end
+              values = value_matcher attribute
+              next if values.nil?
```

Comment:
when the value is either nil or array, I prefer just to cast it to `to_a`
`next if values.to_a.all?`
also I would add a helper method `static?` which encapsulates recursive_literal? || const_type? check